README.md
 The game is written in html and javascript.
 It is designed for nokia s60v5.
 
 ![screenshot](https://raw.githubusercontent.com/roidayan/blackjackwrt/master/assets/Blackjackwrt_1_17923.jpg)

 
Changelog:
 -----Mar 2, 2009 - v0.6
 -----Preloading images to avoid flickering.
 Changed betting style.
   You can see your balance and bet amount.
   Still cant see chips values.
 

-----
Feb 28, 2009 - v0.52
 -----Fixed changelog dates.
 Fixed red cards corners to look like the black cards.
 Moved info button down the screen in portrait mode not to hide house cards.
 Hidden overflow for card images passing screen size.
 Added Double button to double the bet on first initial cards.
 

-----
Feb 28, 2009 - v0.51 - fixes only.
 -----Fixed screen width.
 Fix: House didnt hit a card if less than 17 but higher than player.
 Fixed landscape mode.
 

-----
Feb 28, 2009 - v0.5 - first release.
 -----Missing play options: split, double.
 Needs a better/faster way of adding betting chips.
 Missing chips value.
 Missing how many chips player has.
 Missing landscape mode.
 Save player chips and load when returning to game.
   When out of chips show a message to player and start with random chips.
   Count number of games player is playing and number of wins.


Blackjack/Changelog.txt
-----
Mar 2, 2009 - v0.6
-----
Preloading images to avoid flickering.
Changed betting style.
  You can see your balance and bet amount.
  Still cant see chips values.


-----
Feb 28, 2009 - v0.52
-----
Fixed changelog dates.
Fixed red cards corners to look like the black cards.
Moved info button down the screen in portrait mode not to hide house cards.
Hidden overflow for card images passing screen size.
Added Double button to double the bet on first initial cards.


-----
Feb 28, 2009 - v0.51 - fixes only.
-----
Fixed screen width.
Fix: House didnt hit a card if less than 17 but higher than player.
Fixed landscape mode.


-----
Feb 28, 2009 - v0.5 - first release.
-----
Missing play options: split, double.
Needs a better/faster way of adding betting chips.
Missing chips value.
Missing how many chips player has.
Missing landscape mode.
Save player chips and load when returning to game.
  When out of chips show a message to player and start with random chips.
  Count number of games player is playing and number of wins.


Blackjack/cards.js

//Prepare playing cards and shuffle
function initCards()
{
	//Cards=[];
	for (var d=0; d<NUMBER_OF_DECKS; d++)
		addDeck();
	shuffleCards();
	//for(var i=0; i<90; i++)
	//		Cards.pop();
	console.log("initCards: "+Cards.length);
}

//Add another shuffled deck of cards
function addDeck()
{
	for (var s=0; s<SUITE.length; s++)
		for (var c=0; c<CARDS_IN_SUITE; c++)
			Cards.push(s+':'+(c+1));
		shuffleCards();
}

function swapCards(a,b)
{
	var tmp=Cards[a];
	Cards[a]=Cards[b];
	Cards[b]=tmp;
}

//Shuffle playing cards
function shuffleCards()
{
	for (var n=0; n<Cards.length; n++)
		swapCards(Math.floor(Math.random()*Cards.length),n);
}

//Return filename for card 'suite:number'
function fnCard(card)
{
	var x = card.split(':');
	if (x[0] < 0 || x[0] >= SUITE.length || x[1] < 1 || x[1] > CARDS_IN_SUITE)
		return;
	var fn = "images/"+SUITE[x[0]]+'s/'+SUITE[x[0]]+'_'+CARD[x[1]-1]+CARD_EXT;
	return fn;
}

//Create new card image object
function newCardImage(card,id)
{
	var c = new Image();
	var src = fnCard(card);
	c.src = src;
	c.id = id;
	c.className = 'card';
	return c;
}

//Draw card on board
function drawCard(whom, hidecard)
{
	if (whom != HOUSE && whom != PLAYER)
		return;
	if (Cards.length==0)
		initCards();
		//msgPlayer("Shuffled a new deck.");
	var card = Cards.pop();
	if (!card)//we shouldnt get here
		throw("Error, undefined card. length="+Cards.length);
	//console.log("drawCard: "+card+" (left: "+Cards.length+")");
	var pos = whom.cards.length * CARD_LEFT_POSITION;
	whom.cards.push(card);
	var c = newCardImage(card,whom.id+"_card"+whom.cards.length);
	if (hidecard==HIDDEN_CARD)
		c.src = "images/"+BACK_CARD;
	c.setAttribute("style","left:"+pos+"px;");
	//hitCard doesnt check if player placed a bet.
	//if (whom==PLAYER)
	//	c.setAttribute("onclick","hitCard();");
	var h = document.getElementById(whom.id+'_cards');
	h.appendChild(c);
	sumCards(whom);
}

//Delete cards from board
function remCards(whom)
{
	if (whom != HOUSE && whom != PLAYER)
		return;
	var t = document.getElementById(whom.id+'_cards');
	while (t.childNodes.length > 1)
		t.removeChild(t.lastChild);
}

//Delete all cards from board
function remAllCards()
{
	remCards(HOUSE);
	remCards(PLAYER);
}


Blackjack/chips.js

//Random chips for player
function initPlayerChips()
{
	var sum = 0;
	for (var n=0; n<CHIPS.length; n++)
		sum += Math.floor(Math.random()*400+100);
	payToPlayer(sum);
}

//Pay to player
function payToPlayer(sum)
{
	PLAYER.sumChips += sum;
	BALANCEVIEW.innerHTML = PLAYER.sumChips;
}

//Take chip value from player
function takeChipFromPlayer(chip)
{
	if (PLAYER.sumChips < CHIP_VALUE[chip])
		return false;
	payToPlayer(-1*CHIP_VALUE[chip]);
	return true;
}

//Add chip value to bet
function addChipToBet(chip)
{
	PLAYER.bet.sumChips += CHIP_VALUE[chip];
	BETSVIEW.innerHTML = PLAYER.bet.sumChips;
}

//Return filename for chip
function fnChip(chip)
{
	if (!(chip in CHIP_VALUE))
		return;
	var fn = "images/Chips/"+chip+".gif";
	return fn;
}

//Create new chip image object
function newChipImage(chip,id)
{
	var c = new Image();
	c.src = fnChip(chip);
	c.id = id;
	c.className = 'chip';
	return c;
}

//Draw player betting chip on board
function drawBetChip(c)
{
	var im = newChipImage(c,"betting_chip_"+c);
	if (BETTINGCHIPSVIEW.childNodes.length>1) {//>1 because of the comment
		var top = parseInt(BETTINGCHIPSVIEW.lastChild.style.top);
		var left = parseInt(BETTINGCHIPSVIEW.lastChild.style.left) + CHIP_LEFT_POSITION;
	} else {
		var top=0;
		var left=0;
	}
	//CHIP_ENDLEFT_POSITION
	if (left >= BETTINGCHIPSVIEW.scrollWidth) {
		left=0;
		top += CHIP_TOP_POSITION;
	}
	im.setAttribute("style","left:"+left+"px; top:"+top+"px;");
	BETTINGCHIPSVIEW.appendChild(im);
}

//Draw betting chips according to bets
function drawBetChips()
{
	var sum = PLAYER.bet.sumChips;
	var c;
	for (var n=CHIPS.length-1; n>=0; n--) {
		c=CHIPS[n];
		while (sum >= CHIP_VALUE[c]) {
			drawBetChip(c);
			sum -= CHIP_VALUE[c];
		}
	}
}

//Delete betting/player chips from board
function remChips(what)
{
	if (what!=BETTINGCHIPSVIEW && what!=PLAYERCHIPSVIEW)
		return;
	while (what.childNodes.length > 1)
		what.removeChild(what.lastChild);
}

//ReDraw betting/player chips on board and sum the chips
function refreshChipsView(what)
{
	if (what!=BETTINGCHIPSVIEW && what!=PLAYERCHIPSVIEW)
		return;
	remChips(what);
	if (what==BETTINGCHIPSVIEW)
		drawBetChips();
	else if (what==PLAYERCHIPSVIEW)
		drawPlayerChips();
}

//Clear player bet chips from board
function clearBet()
{
	PLAYER.bet.sumChips=0;
	BETSVIEW.innerHTML=0;
	remChips(BETTINGCHIPSVIEW);
}

//Return bets to player
function returnBet()
{
	payToPlayer(PLAYER.bet.sumChips);
	clearBet();
}

//Draw chip player has on board
function drawPlayerChip(chip)
{
	var im = newChipImage(chip,"player_chip_"+chip);
	im.setAttribute("style","position:relative;");
	im.setAttribute("onclick","placeBet('"+chip+"');");
	//im.setAttribute("onmousedown","placeBet('"+chip+"');");
	//im.setAttribute("onmouseup","stopBet();");
	//im.setAttribute("onmouseout","stopBet();");
	//PLAYERCHIPSVIEW.appendChild(im);
	PLAYERCHIPSVIEW.appendChild(im);
}

//Draw chips player can use on board
function drawPlayerChips()
{
	for (var c in CHIP_VALUE)
		if (PLAYER.sumChips >= CHIP_VALUE[c])
			drawPlayerChip(c);
}

//Player placing a bet - OnMouseDown
function placeBet(chip)
{
	if (!takeChipFromPlayer(chip))
		return;
	vbrbasic();
	//drawBetChip(chip);
	addChipToBet(chip);
	//if (!PLAYER.chips[chip])
		//alert(document.getElementById("player_chip_"+chip));
		//PLAYERCHIPSVIEW.removeChild(document.getElementById("player_chip_"+chip));
	//PlaceBetTimeout = setTimeout("placeBet('"+chip+"');",BET_DELAY);
}
//OnMouseUp
/*function stopBet()
{
	clearTimeout(PlaceBetTimeout);
}*/

//Pay player his bets ratio
function payBet()
{
	var sum = PLAYER.bet.sumChips;
	if (PLAYER.sumCards==21)
		sum = Math.floor( ( sum*PAY_RATIO_BLACKJACK[0] ) / PAY_RATIO_BLACKJACK[1] );
	else
		sum = sum*2;
	payToPlayer(sum);
	clearBet();
}

//Show betting chips on board
function showBettingChips()
{
	PLAYERCHIPSVIEW.style.display = 'none';
	BETTINGCHIPSVIEW.style.display = 'inline';
	refreshChipsView(BETTINGCHIPSVIEW);
}

//Show player chips on board
function showPlayerChips()
{
	PLAYERCHIPSVIEW.style.display = 'inline';
	BETTINGCHIPSVIEW.style.display = 'none';
	refreshChipsView(PLAYERCHIPSVIEW);
}


Blackjack/bj.js
/**
	Blackjack by Roi Dayan
**/

//GLOBALS
var VERSION = "0.6";

//CARDS
var SUITE =  [ 'Club', 'Spade', 'Diamond', 'Heart' ];
var CARDS_IN_SUITE = 13;
var CARDS_IN_DECK = SUITE.length * CARDS_IN_SUITE;
var CARD = [ 'ace', 2,3,4,5,6,7,8,9,10, 'jack', 'queen', 'king' ];
var CARD_EXT = "_75x110.png";
var BACK_CARD = "card_background"+CARD_EXT;

var NUMBER_OF_DECKS = 2;
var TOTAL_CARDS = NUMBER_OF_DECKS * CARDS_IN_DECK;
var Cards=[];					//Hold shuffeld cards as 'suite:number'

var CARD_LEFT_POSITION = 40;	//Draw card multiply position

//CHIPS
var CHIPS = [ 'Purple', 'Yellow', 'Red', 'Orange', 'Green', 'Blue', 'Black' ];
var CHIP_VALUE = {'Purple':1, 'Yellow':5, 'Red':25, 'Orange':100, 'Green':200, 'Blue':500, 'Black':1000};
//var CHIP_TOP_POSITION = 15;		//Draw chip multiply top position
//var CHIP_LEFT_POSITION = 10;	//Draw chip multiply left position
var CHIP_TOP_POSITION = 20;		//Draw chip multiply top position
var CHIP_LEFT_POSITION = 20;	//Draw chip multiply left position
var CHIP_ENDLEFT_POSITION;		//Is set according to css #betting_chips width.

var BET_DELAY = 100;
var PlaceBetTimeout;			//Set OnMouseDown and clear OnMouseUp

//STRUCTURES
var HOUSE = {id:'house', cards:[], sumCards:0};
//chips are dictionary of total chips player has of chip:count
//bet dictionary is chips player is betting
var PLAYER = {id:'player', cards:[], sumCards:0, chips:{}, sumChips:0, bet:{chips:{}, sumChips:0}};

//GAME
var GAME_DELAY = 100;
var NO = 0;
var YES = 1;
var PUSH = 2;
var BET = 3;
var Play = NO;
var HIDDEN_CARD = 1;
//var PAY_RATIO = [1,1]; //1 to 1
var PAY_RATIO_BLACKJACK = [3,2]; //3 to 2

/********************************************/

//ELEMENTS
var BODY, MAIN;
var GAMEVIEW, BETTINGCHIPSSVIEW, CHIPSVIEW, PLAYERCHIPSVIEW;
var BUTTON_DEAL, BUTTON_HIT, BUTTON_BET, BUTTON_STAND, BUTTON_DOUBLE;
var MSGVIEW, BALANCEVIEW, BETSVIEW;
//player_sum, house_sum

//PHONE
var sysInfo=null;	//SystemInfo Object
var PORTRAIT = "portrait";
var HORIZONTAL = "horizontal";

function init()
{
	if (!ImagesPreloader.check()) {
		setTimeout("init();",500);
		return;
	}

	//init elements
	BODY = document.getElementById('body');
	MAIN = document.getElementById('main');
	GAMEVIEW = document.getElementById('game');
	BETSVIEW = document.getElementById('bets');
	BETTINGCHIPSVIEW = document.getElementById('betting_chips');
	PLAYERCHIPSVIEW = document.getElementById('player_chips');
	BUTTON_DEAL = document.getElementById('button-deal');
	BUTTON_HIT = document.getElementById('button-hit');
	BUTTON_BET = document.getElementById('button-bet');
	BUTTON_STAND = document.getElementById('button-stand');
	BUTTON_DOUBLE = document.getElementById('button-double');
	MSGVIEW = document.getElementById('msg');
	BALANCEVIEW = document.getElementById('balance');
	BETSVIEW = document.getElementById('bet');

	initPhone();

	initPlayerChips();

	Hold("fade");
		document.getElementById('header').style.display = "none";
		GAMEVIEW.style.display = "inline";
	Update();

	//outScreen(HORIZONTAL);
	startGame();
}

//init phone settings
function initPhone()
{
	if (!window.widget)
		return;

	window.menu.hideSoftkeys();

	//getDisplayOrientation();
	setInterval("getDisplayOrientation()",500);

	// Obtain the SystemInfo object
    try {
        sysInfo = document.embeds[0];
    } catch (ex) {
        alert("SystemInfo object cannot be found.");
        return;
    }

}

// Display Information
function getDisplayOrientation()
{
    //if (widget.isrotationsupported)

	// Change the screen orientation
	//widget.setDisplayLanscape();

	var h = window.screen.height;
	var w = window.screen.width;
	
	if (h > w)
		outScreen(PORTRAIT);
  	else
		outScreen(HORIZONTAL);
}

//Phone vibration
function vbr(duration,intensity) {
	if (sysInfo==null)
		return;
	durationvalue = Number(duration);
	intensityvalue = Number(intensity);
	sysInfo.startvibra(durationvalue, intensityvalue);
}
function vbrbasic() {
	if (sysInfo==null)
		return;
	sysInfo.startvibra(18,10);
}

//Widget transition
function Hold(style)
{
	if (sysInfo==null)
		return;
	widget.prepareForTransition(style);
}
function Update()
{
	if (sysInfo==null)
		return;
	setTimeout("widget.performTransition();", 0);
}

//Show cards if player has a bet, else shows the chips.
function refreshView()
{
	updateButtons();
	if (Play==BET)
		showPlayerChips();
	else if (Play==YES && PLAYER.bet.sumChips > 0) {
		showBettingChips();
		showCard(HOUSE,0);
		setTimeout("showCards(PLAYER); checkPlayerCards(); updateButtons(1);",GAME_DELAY);
	}
}

//Update game buttons status
//If initial==YES it means showing first cards so to allow double.
function updateButtons(initial)
{
	//if ((Play==YES && !initial) || (Play==PUSH)) {
	//if (Play==YES && !initial) 
	if (Play==BET) {
		showButton(BUTTON_BET);
		hideButton(BUTTON_DOUBLE);
	} else if (initial) {
		hideButton(BUTTON_BET);
		showButton(BUTTON_DOUBLE);
	} else {
		BUTTON_BET.disabled = true;
		BUTTON_DOUBLE.disabled = true;
	}
//	if (Play==PUSH)
//		return;
	if (Play==YES && PLAYER.bet.sumChips > 0) {
		BUTTON_HIT.disabled = false;
		BUTTON_STAND.disabled = false;
	} else {
		BUTTON_HIT.disabled = true;
		BUTTON_STAND.disabled = true;
	}
}

//Show and enable button
function showButton(btn)
{
		btn.disabled = false;
		btn.style.display = 'inline';
}
//Hide and disable button
function hideButton(btn)
{
	btn.disabled = true;
	btn.style.display = 'none';
}

//Sum cards according to rules
function sumCards(whom)
{
	if (whom != HOUSE && whom != PLAYER)
		return;
	var sum = 0;
	var aces = 0;
	var c;
	for (var i in whom.cards)
	{
		c = parseInt(whom.cards[i].split(':')[1]);
		if (c>10)
			c=10;
		else if (c==1)
			aces++;
		sum += c;
		if (c==1)
			sum+=10;
	}
	
	while (sum>21 && aces>0)
	{
		aces--;
		sum-=10;
	}
	
	document.getElementById(whom.id+'_sum').innerHTML = sum;
	whom.sumCards = sum;
}

//Hit a card - draw and check
function hitCard()
{
	if (Play==NO)
		return;
	vbrbasic();
	BUTTON_BET.disabled = true;
	BUTTON_DOUBLE.disabled = true;
	drawCard(PLAYER);
	checkPlayerCards();
}

//Show a message for the player
function msgPlayer(msg)
{
	MSGVIEW.innerHTML = msg;
	MSGVIEW.style.display = 'inline';
}

//Finnish game
function finishGame()
{
	updateScore();
	//Play=NO;
	//if (Play==PUSH)
		//
	refreshView();
	showButton(BUTTON_DEAL);
}

//Player is standing
function Stand()
{
	if (Play==NO)
		return;
	Play=NO;
	vbrbasic();
	updateButtons();
	if (PLAYER.sumCards>21)
		finishGame();
	else {
		setTimeout("showCard(HOUSE,1); document.getElementById(HOUSE.id+'_sum').style.display = 'inline';",GAME_DELAY);
		setTimeout("housePlay();",GAME_DELAY);
	}
}

//House is playing, assumption Play=NO
//TODO: done here?
function housePlay()
{
	if (HOUSE.sumCards<17)
	{
		setTimeout("drawCard(HOUSE); housePlay()",GAME_DELAY);
		return;
	}
	
	//can add soft17 mode here.

	finishGame();
}

//Update score and player message
function updateScore()
{
	var win = NO;

	if (PLAYER.sumCards > 21)
		msgPlayer("Busted");
	else if (HOUSE.sumCards > 21) {
		msgPlayer("House Busted");
		win=YES;
	} else if (HOUSE.sumCards==PLAYER.sumCards) {
		msgPlayer("Push");
		win=PUSH;
	} else if (HOUSE.sumCards < PLAYER.sumCards) {
		msgPlayer("You Win");
		win=YES;
	} else if (HOUSE.sumCards > PLAYER.sumCards)
		msgPlayer("You Lost");
	
	if (win==NO) {
		clearBet();
		return;
	} else if (win==PUSH) {
		Play=PUSH;
		return;
	}
	
	if (PLAYER.sumCards==21)
		msgPlayer("Blackjack");
	
	payBet();
}

//Deal cards for new game
function dealCards()
{
	vbrbasic();
	//TODO: i dont think i'm done here
	Play=BET;
	hideButton(BUTTON_DEAL);
	MSGVIEW.style.display = 'none';
	resetPlayersCards();
	startGame();
}

//Reset player and house cards information
function resetPlayersCards()
{
	HOUSE.cards=[];
	HOUSE.sumCards=0;
	PLAYER.cards=[];
	PLAYER.sumCards=0;
	remAllCards();
}

//Start game (assuming new game) - deal house one card, deal player 2 cards
function startGame()
{
	Play=NO;
	refreshView();
	document.getElementById(HOUSE.id+'_sum').style.display = 'none';
	document.getElementById(PLAYER.id+'_sum').style.display = 'none';
	setTimeout("drawCard(HOUSE,HIDDEN_CARD)",GAME_DELAY);
	setTimeout("drawCard(HOUSE,HIDDEN_CARD)",2*GAME_DELAY);
	setTimeout("drawCard(PLAYER,HIDDEN_CARD)",3*GAME_DELAY);
	setTimeout("drawCard(PLAYER,HIDDEN_CARD); Play=BET; refreshView();",4*GAME_DELAY);
}

//Player decided on bet, continue playing.
function setBet()
{
	if (PLAYER.bet.sumChips==0 || Play!=BET)
		return;
	Play=YES;
	refreshView();
}

//Player wants to clear the bet
function resetBet()
{
	if (Play!=BET)
		return;
	returnBet()
	refreshChipsView(PLAYERCHIPSVIEW);
}

//Double the bet
function doubleBet()
{
	vbrbasic();
	if (PLAYER.sumChips < PLAYER.bet.sumChips)
	{
		alert("You do not have enough chips, ("+PLAYER.bet.sumChips+"<"+PLAYER.sumChips+").");
		return;
	}
	//
	var sum = PLAYER.bet.sumChips;
	var c;
	for (var n=CHIPS.length-1; n>=0 && sum>0; n--)
	{
		c=CHIPS[n];
		while (sum>0 && sum>=CHIP_VALUE[c] && takeChipFromPlayer(c)) {
			drawBetChip(c);
			addChipToBet(c);
			sum -= CHIP_VALUE[c];
		}
	}
	drawCard(PLAYER);
	checkPlayerCards(1);
}

//Check player cards
function checkPlayerCards(forcestand)
{
	if (PLAYER.sumCards < 21 && !forcestand)
		return;
	if (PLAYER.sumCards==21)
		msgPlayer("Blackjack");
	Stand();
}

//Hide cards
function hideCard(whom,index)
{
	document.images[whom.id+'_card'+(index+1)].src = "images/"+BACK_CARD;
}
function hideCards(whom)
{
	if (whom != HOUSE && whom != PLAYER)
		return;
	for (var n=0; n<whom.cards.length; n++)
		hideCard(whom,n);
		//document.images[whom.id+'_card'+(n+1)].src = "images/"+BACK_CARD;
}

//Show cards
function showCard(whom,index)
{
	document.images[whom.id+'_card'+(index+1)].src = fnCard(whom.cards[index]);
}
function showCards(whom)
{
	if (whom != HOUSE && whom != PLAYER)
		return;
	for (var n=0; n<whom.cards.length; n++)
		showCard(whom,n);
		//document.images[whom.id+'_card'+(n+1)].src = fnCard(whom.cards[n]);
	document.getElementById(whom.id+'_sum').style.display = 'inline';
}

//Change output screen between horizontal and portrait
function outScreen(t)
{
	if (t == BODY.className)
		return;

	BODY.className = t;
	MAIN.className = t;
}

//Show about information
function showInfo()
{
	vbrbasic();
	Hold("fade");
		GAMEVIEW.style.display = "none";
		document.getElementById('info').style.display = "inline";
	Update();
}
//Close about information
function closeInfo()
{
	vbrbasic();
	Hold("fade");
		GAMEVIEW.style.display = "inline";
		document.getElementById('info').style.display = "none";
	Update();
}

//About information text
function about()
{
	document.writeln("Blackjack v"+VERSION+"<br>by Roi Dayan<br><br>");
	document.writeln("<span style='font-size:20px;'>http://blackjackwrt.googlecode.com<br><br>Click here to continue</span>");
}


/*
//Cookies

function setCookie(c_name,value,expiredays)
{
	var exdate = new Date();
	exdate.setDate(exdate.getDate()+expiredays);
	document.cookie = c_name+"="+escape(value)+((expiredays==null) ? "" : ";expires="+exdate.toGMTString());
}

function getCookie(c_name)
{
	if (document.cookie.length>0)
	{
		c_start = document.cookie.indexOf(c_name+'=');
		if (c_start!=-1)
		{ 
			c_start = c_start+c_name.length+1; 
			c_end = document.cookie.indexOf(";",c_start);
			if (c_end==-1) c_end = document.cookie.length;
			return unescape(document.cookie.substring(c_start,c_end));
		}
	}
	return "";
}

var tmp="BlackjackWidget";
setCookie(tmp,'a103',100);
alert(getCookie(tmp));
*/

//Images Preloader
var ImagesPreloader = {
	list: [],
	images: [],

	add: function(src) {
		this.list = this.list.concat(src);
	},

	start: function(src) {
		var im;
		for (var i=0; i<this.list.length; i++) {
			im = new Image();
			im.src = src;
			this.images.push(im);
		}
	},

	check: function() {
		for (var i=0; i<this.images.length; i++)
			if (!this.images[i].complete)
				return false;
		return true;
	}
}
//Preload images
function preloadImages()
{
	for (var s=0; s<SUITE.length; s++)
		for (var c=0; c<CARDS_IN_SUITE; c++)
			ImagesPreloader.add("images/"+SUITE[s]+'s/'+SUITE[s]+'_'+CARD[c]+CARD_EXT);
	for (var c in CHIP_VALUE)
		ImagesPreloader.add("images/Chips/"+c+".gif");
	ImagesPreloader.add("images/"+BACK_CARD);
	ImagesPreloader.add("images/deskh.PNG");
	ImagesPreloader.add("images/deskp.PNG");
	ImagesPreloader.add("images/Info.gif");
	ImagesPreloader.add("images/Close.gif");
	ImagesPreloader.start();
}
